node/node.go

```diff
@@ -230,14 +229,13 @@ func NewNode(config *cfg.Config,
 		}
 	}
 
-	// Decide whether to fast-sync or not
-	// We don't fast-sync when the only validator is us.
+	// Decide whether to fast-sync or not.
+	// We don't fast-sync when we have +2/3 of the validator set.
 	fastSync := config.FastSync
-	if state.Validators.Size() == 1 {
-		addr, _ := state.Validators.GetByIndex(0)
-		if bytes.Equal(privValidator.GetAddress(), addr) {
-			fastSync = false
-		}
+	ourAddr := privValidator.GetAddress()
+	_, val := state.Validators.GetByAddress(ourAddr)
+	if val != nil && val.VotingPower > state.Validators.TotalVotingPower()*2/3 {
```

Comment:
This is another case where we require total voting power to be smaller than its max value. (In this case less than 2**62, previous mention of need for a requirement: https://github.com/tendermint/tendermint/pull/2785#issuecomment-440357821)